web/src/unread.js

```diff
@@ -281,6 +281,38 @@ class UnreadTopicCounter {
         return res;
     }
 
+    get_topics_count(include_muted = false) {
```

Comment:
I don't think it matters how many unreads there are really for how fast the updates are. The number of unread topics is the real concern. Updates with 25k unreads were fast enough. Like 1s to mark a stream with 2k unreads as read and update the UI.

I am not sure we can combine them. I added a comment.

Reply:
Yeah maybe the data structures here are fast enough (especially compared to HTML rendering) that doing all the work to extract the data from the data structures twice is not a performance problem.

I think it's still a correctness risk to have two copies of almost the same function, but it's potentially a risk we can live with for now. Maybe extend the comment with "TODO: Refactor so that we don't need two versions of this."

Probably the look of that refactor would be to just extend the original function to return additional values for the alternative rules.

Reply:
Makes sense. Added that comment.